LeaderboardBackend.Test/Leaderboards.cs

```diff
@@ -348,4 +349,190 @@ public async Task GetLeaderboards()
         LeaderboardViewModel[] returned = await _apiClient.Get<LeaderboardViewModel[]>("/api/leaderboards", new());
         returned.Should().BeEquivalentTo(boards.Take(2), config => config.Excluding(lb => lb.Categories));
     }
+
+    [Test]
+    public async Task RestoreLeaderboard_OK()
+    {
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        Leaderboard deletedBoard = new()
+        {
+            Name = "Super Mario World",
+            Slug = "super-mario-world-to-restore",
+            DeletedAt = _clock.GetCurrentInstant()
+        };
+
+        context.Leaderboards.Add(deletedBoard);
+        await context.SaveChangesAsync();
+        deletedBoard.Id.Should().NotBe(default);
+
+        _clock.AdvanceMinutes(1);
+
+        LeaderboardViewModel res = await _apiClient.Put<LeaderboardViewModel>($"/leaderboard/{deletedBoard.Id}/restore", new()
+        {
+            Jwt = _jwt
+        });
+
+        res.Id.Should().Be(deletedBoard.Id);
+        res.Slug.Should().Be(deletedBoard.Slug);
+        res.UpdatedAt.Should().Be(_clock.GetCurrentInstant());
+        res.DeletedAt.Should().BeNull();
+    }
+
+    [Test]
+    public async Task RestoreLeaderboard_Unauthenticated()
+    {
+        Func<Task<LeaderboardViewModel>> act = async () => await _apiClient.Put<LeaderboardViewModel>($"/leaderboard/100/restore", new());
+
+        await act.Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.Unauthorized);
+    }
+
+    [Test]
+    public async Task RestoreLeaderboard_Banned_Unauthorized()
+    {
+        string email = "restore-leaderboard-banned@example.com";
+        string password = "P4ssword";
+
+        UserViewModel userModel = await _apiClient.RegisterUser(
+            "RestoreBoardBanned",
+            email,
+            password
+        );
+
+        string jwt = (await _apiClient.LoginUser(email, password)).Token;
+
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        context.Users.Update(new()
+        {
+            Id = userModel.Id,
+            Role = UserRole.Banned,
+            Username = userModel.Username,
+            Email = email,
+            Password = password
```

Comment:
You're actually unintentionally overwriting the user's salted and hashed password with a plaintext password here. It's not causing an issue right now because of the password you've chosen but it could in the future. Do something like `context.Users.Single(user => user.Id == userModel.Id).Role = UserRole.Banned;` instead.

Reply:
Remind me to not write stuff when I'm tired